### DISCUSSION TITLE
Code snippet plugin always empty

### DISCUSSION THREAD

**Discussion by Nixinova** [lowlighter/metrics#870] [Q&A]
I enabled the code snippet plugin a month ago but I've never had it show anything. The only option ive used is `plugin_code: yes`. How do I get the plugin to show anything? Are there only specific types of commits it looks for or something?

**Comment by Nixinova** [comment]
Well, just now it has started showing content, but that content is from Jan 6 : https://github.com/Nixinova/Nixinova/commit/a8b9d4082eeee7a500af5cc95aeb5438317170a2. Does it ignore the last month or something?
![image](https://user-images.githubusercontent.com/42429413/153669035-826a86cf-5310-4f8b-be5b-7df6728b2edf.png)

**Reply by lowlighter** [reply]
The `plugin_code_load` option is quite low (defaults at 100) so it's easily possible to not have a push event (since reviews, prs, issues, comments, forks, etc... count towards limit)

As for how a snippet is chosen, it'll search a single file from a patch with less than `plugin_code_lines` lines (defaults 12). This is to avoid getting a random a file with a lot of editions taking the whole space. So basically any diff with more than 12 lines (including diff headers and common lines) will be skipped if you use the different config

**Reply by Nixinova** [reply]
> The plugin_code_load option is quite low (defaults at 100) so it's easily possible to not have a push event (since reviews, prs, issues, comments, forks, etc... count towards limit)

But what seems to be happening is its only loading *old* commits. the ones shown are from early Jan over a month ago

**Reply by lowlighter** [reply]
Hum this is strange indeed, will look into it later next week 👍 
I think it's a good opportunity to add an option to restrict the time window since other plugins have it but not this one
